WcaOnRails/app/helpers/application_helper.rb

```diff
@@ -177,4 +177,12 @@ def duration_to_s(total_seconds)
 
     [hours > 0 ? "#{hours}h " : '', minutes > 0 ? "#{minutes}m " : '', format('%.2f', seconds), 's'].join
   end
+
+  def wca_id_link(wca_id, options = {})
+    if wca_id
+      content_tag :span, class: "wca-id" do
+        link_to wca_id, "#{root_url}results/p.php?i=#{wca_id}", options
```

Comment:
Cool! 

I'd still like a`:relative` option for the non email cases. This line should do it:

`root = options.delete(:relative) ? '/' : root_url`